### PR TITLE
[MXNET-833] [R] Char-level RNN tutorial fix

### PR DESCRIPTION
**Pull request by anirudhacharya**
## Description ##
Char-level RNN model for language modeling.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by vandanavk** [comment]
@mxnet-label-bot [pr-awaiting-review]

File: R-package/vignettes/CharRnnModel.Rmd [hunk 1]
```diff
@@ -36,237 +32,262 @@ download.data <- function(data_dir) {
...
+  text_vec <- strsplit(text_vec, '') %>% unlist
+  
+  if (is.null(dic)) {
+    char_keep <- sort(unique(text_vec))
+  } else char_keep <- names(dic)[!dic == 0]
+  
+  # Remove terms not part of dictionnary
+  text_vec <- text_vec[text_vec %in% char_keep]
+  
+  # Build dictionnary
```
Comment (ankkhedia) [inline R-package/vignettes/CharRnnModel.Rmd]: typo in dictionary

File: R-package/vignettes/CharRnnModel.Rmd [hunk 2]
```diff
@@ -36,237 +32,262 @@ download.data <- function(data_dir) {
...
-    else {
-        idx <- choice(prob)
-    }
-    return (idx)
-
+## Train the Model
+
+
+This model is a multi-layer RNN for sampling from character-level language models. It has a one-to-one model configuration since for each character, we want to predict the next one. For a sequence of length 100, there are also 100 labels, corresponding the same sequence of characters but offset by a position of +1. The parameters output_last_state is set to TRUE in order to access the state of the RNN cells when performing inference.
+
```
Comment (ankkhedia) [inline R-package/vignettes/CharRnnModel.Rmd]: Would it be helpful if we mention that currently only one-to-one and many-to-one flavours are supported.

Comment (anirudhacharya) [reply R-package/vignettes/CharRnnModel.Rmd]: it would not be relevant to this tutorial

File: R-package/vignettes/CharRnnModel.Rmd (hunk 1 above)
Comment (anirudhacharya) [reply R-package/vignettes/CharRnnModel.Rmd]: will fix

File: R-package/vignettes/CharRnnModel.Rmd [hunk 3]
```diff
@@ -36,237 +32,262 @@ download.data <- function(data_dir) {
...
+  # Build dictionnary
+  dic <- 1:length(char_keep)
+  names(dic) <- char_keep
+  
+  # reverse dictionnary
+  rev_dic <- names(dic)
+  names(rev_dic) <- dic
+  
+  # Adjust by -1 to have a 1-lag for labels
+  num.seq <- as.integer(floor((length(text_vec)-1)/seq.len))
```
Comment (hetong007) [inline R-package/vignettes/CharRnnModel.Rmd]: `a %/% b` gives you integer division.

**Comment by sandeep-krishnamurthy** [comment]
@ankkhedia - Your comments are addressed. Can you please take a look? Thanks.

**Comment by vandanavk** [comment]
@sandeep-krishnamurthy This PR is ready for merge

**Pull request merged**